group/assignment6/Assignment6.cpp

```diff
@@ -0,0 +1,114 @@
+#include <iostream>
+#include <vector>
+
+int findVacantSlot(const std::vector<int> &parking_lot) {
+	for (int i = 0; i < parking_lot.size();i++) {
+		if (parking_lot[i] == -1)
+			return i;
+	}
+}
+std::vector<int> getCarsWrongPlace(const std::vector<int> &starting_state, const std::vector<int> &ending_state) {
+	std::vector<int> cars_wrong_place;
+	for (int i = 0; i < starting_state.size(); i++) {
+		if (starting_state[i] != ending_state[i] && starting_state[i] != -1)
+			cars_wrong_place.push_back(starting_state[i]);
+	}
+	return cars_wrong_place;
+}
+int findCarSlot(const std::vector<int> &parking_lot, int car) {
+	for (int i = 0; i < parking_lot.size(); i++) {
```

Comment:
I'm not sure, but can it be better to use a hashmap for the starting_state? But this requires additional memory and a little more complicated logic :c

Reply:
I agree on this. It will improve the time complexity. We ll have to do it once and then just access it in O(1) time